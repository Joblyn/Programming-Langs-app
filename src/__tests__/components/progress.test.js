import React from "react";
import { render, cleanup } from "@testing-library/react";
import ProgressLine from "../../components/progress/progressline";

describe("<Progress />", () => {
  afterEach(cleanup);

  // UI test
  it("renders component correctly", () => {
    const count = 1;
    const totalcount = 15;

    const { container, getByText } = render(
      <ProgressLine count={count} totalcount={totalcount} />
    );

    expect(getByText(`${count}/${totalcount}`)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
