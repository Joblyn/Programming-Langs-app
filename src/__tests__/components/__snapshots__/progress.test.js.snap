// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> renders component correctly 1`] = `
<div
  class="MuiBox-root css-8atqhb"
>
  <div
    class="MuiBox-root css-13buc9q"
  >
    <div
      class="MuiBox-root css-1bhivts"
    >
      <span
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="7"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-eglki6-MuiLinearProgress-root"
        count="1"
        role="progressbar"
        totalcount="15"
      >
        <span
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
          style="transform: translateX(-93%);"
        />
      </span>
    </div>
    <div
      class="MuiBox-root css-pazuet"
    >
      <span
        class="ml-auto text-xl"
      >
        1
        /
        15
      </span>
    </div>
  </div>
</div>
`;
